test/cache_test.rb

```diff
@@ -4,6 +4,26 @@
 
 module ActiveModelSerializers
   class CacheTest < ActiveSupport::TestCase
+    class AuthorWithCustomCacheKey < Author
+      class DerivedCacheKey
+        def initialize(object, method)
+          @object = object
+          @method = method
+        end
+
+        def cache_key
+          @object.__send__(@method)
+        end
+      end
```

Comment:
Maybe

```ruby
class AuthorWithCustomCacheKey < Author
  HasCacheKey = Struct.new(:cache_key)
  
  # Rather than `[name, id]` as the cache key, demonstrate that `:cache_key` is called on
  # any element that responds to it.
  def cache_key
    [HasCacheKey.new(name), HasCacheKey.new(id)]
  end
end
```